### PR TITLE
[RFR] Implement full(Request|Response)Interceptors to be able to alter url, params, headers and/or data

### PR DESCRIPTION
**Pull request by jeromemacias**
* [x] Implement `addFullRequestInterceptor`
* [x] Implement `addFullResponseInterceptor`
* [x] Add doc

### CODE REVIEW

**Comment by fzaninotto** [comment]
:+1:

**Comment by jpetitcolas** [comment]
@jeromemacias, still in WIP?

**Comment by jeromemacias** [comment]
@jpetitcolas Missing documentation, will write it soon.

**Comment by jeromemacias** [comment]
@jpetitcolas RFR

File: .jshintrc [hunk 1]
```diff
@@ -2,5 +2,7 @@
     "curly": true,
     "eqeqeq": true,
     "undef": true,
-    "esnext": true
+    "esnext": true,
+    "devel": true,
```
Comment (jpetitcolas) [inline .jshintrc]: I would remove it. This way we'll see at build time if we have some debug in our files.

File: README.md [hunk 2]
```diff
@@ -205,8 +205,10 @@ Restful.js exposes similar methods on collections, members and entities. The nam
 * `patch ( id, data [, headers ] )`: Patch a member in a collection. Returns a promise with the response.
 * `head ( id, [, headers ] )`: Perform a HEAD request on a member in a collection. Returns a promise with the response.
 * `url ()`: Get the collection url.
-* `addResponseInterceptor ( interceptor )`: Add a response interceptor.
+* `addResponseInterceptor ( interceptor )`: Add a response interceptor. You can only alter data and headers.
 * `addRequestInterceptor ( interceptor )`: Add a request interceptor.
+* `addFullResponseInterceptor ( interceptor )`: Add a full response interceptor. You can alter data and headers.
```
Comment (jpetitcolas) [inline README.md]: What is the difference with the simple `addResponseInterceptor` then?

File: README.md [hunk 3]
```diff
@@ -244,12 +248,52 @@ A response or request interceptor is a callback which looks like this:
...
 ```
 
+A full request interceptor is a callback which looks like this:
+
+```js
+resource.addFullRequestInterceptor(function(params, headers, data, method, url) {
+    //...
+
+    // all args had been modified
+    return {
```
Comment (jpetitcolas) [inline README.md]: To ease understanding, I would rather write:

``` js
// just return modified arguments
return {
    headers: headers,
    data: data
}
```

**Comment by jpetitcolas** [comment]
Don't commit the build file to ease merges. Will build it before public release.

**Comment by jeromemacias** [comment]
Yes I needed it to use this branch in my `packages.json` project. As there is, by now, only one open PR on the project, can we merge it ?

I will follow this rule on my next PRs.

File: README.md (hunk 2 above)
Comment (jeromemacias) [inline README.md]: One interceptor is executed by `axios`, the other by the `restful.js` library, and the usage is clearer on the `restful.js` one, no magic modification. 
But in fact, there is no real difference, I wasn't see the comment in the documentation before starting the `fullResponseInterceptor`.

**Comment by jpetitcolas** [comment]
:+1:

File: README.md (hunk 2 above)
Comment (jeromemacias) [inline README.md]: And there there is a json parse in the `responseInterceptor` which is not needed into the `fullResponseInterceptor`.

**Comment by jpetitcolas** [comment]
Releasing 0.7.

**Comment by jpetitcolas** [comment]
Published on NPM.

**Pull request merged**